x-pack/plugins/enterprise_search/public/applications/index.tsx

```diff
@@ -9,38 +9,49 @@ import ReactDOM from 'react-dom';
 import { BrowserRouter, Route, Redirect } from 'react-router-dom';
 
 import { CoreStart, AppMountParams, HttpHandler } from 'src/core/public';
-import { ClientConfigType } from '../plugin';
+import { ClientConfigType, PluginsSetup } from '../plugin';
 import { TSetBreadcrumbs } from './shared/kibana_breadcrumbs';
+import { ILicense } from '../../../../licensing/public';
+import { LicenseProvider } from './shared/licensing';
 
 import { AppSearch } from './app_search';
 
 export interface IKibanaContext {
   enterpriseSearchUrl?: string;
   http(): HttpHandler;
   setBreadCrumbs(): TSetBreadcrumbs;
+  license$: Observable<ILicense>;
 }
 
 export const KibanaContext = React.createContext();
 
-export const renderApp = (core: CoreStart, params: AppMountParams, config: ClientConfigType) => {
+export const renderApp = (
+  core: CoreStart,
+  params: AppMountParams,
+  config: ClientConfigType,
+  plugins: PluginsSetup
+) => {
   ReactDOM.render(
     <KibanaContext.Provider
       value={{
         http: core.http,
         enterpriseSearchUrl: config.host,
         setBreadcrumbs: core.chrome.setBreadcrumbs,
+        license$: plugins.licensing.license$,
       }}
     >
-      <BrowserRouter basename={params.appBasePath}>
-        <Route exact path="/">
-          {/* This will eventually contain an Enterprise Search landing page,
-          and we'll also actually have a /workplace_search route */}
-          <Redirect to="/app_search" />
-        </Route>
-        <Route path="/app_search">
-          <AppSearch />
-        </Route>
-      </BrowserRouter>
+      <LicenseProvider>
```

Comment:
Note that the approach I used was heavily borrowed from the [APM plugin's](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/apm/public/context/LicenseContext/index.tsx) [approach](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/apm/public/application/index.tsx#L89) - I looked at 2-3 other plugins and how they handled licensing but ended up liking theirs as the most simple/cleanest/easiest to test.

I also did briefly play around with munging LicenseProvider into a single KibanaProvider declared within this file, but eventually opted to copy their setup - a separate context file is easier to test and notice the observable subscription in, and it's additionally very possible we might just continue to add contexts over time (although it would also be nice if we could handle that w/ Kea...)

Reply:
Nice, I like that you followed the approach the other plugin's use. That seems to make the most sense to me as well.

We don't have Kea yet, right? This could definitely be ported over at some point whenever we get there. It would probably be cleaner than mixing Contexts + Kea stores.